packages/docusaurus-plugin-content-docs/src/sidebars/generator.ts

```diff
@@ -236,8 +236,11 @@ export const DefaultSidebarItemsGenerator: SidebarItemsGenerator = async ({
         ? {
             type: 'doc',
             id: categoryLinkedDocId, // We "remap" a potentially "local id" to a "qualified id"
+            className:
+              categoryMetadata?.link?.className ??
```

Comment:
we already have `categoryMetadata?.className` and it solves user reported use-case, why introduce a new `categoryMetadata?.link?.className` here? 


You could do this instead:

```tsx
category.className = categoryMetadata?.className ?? getDoc(link.id).frontMatter.sidebar_class_name
```